lkmpg.tex

```diff
@@ -1421,6 +1421,14 @@ \subsection{Tasklets}
 Example tasklet init continues...
 Example tasklet ends
 \end{verbatim}
+Although tasklet is easy to use, it comes with several defators, and developers are discussing about getting rid of tasklet in linux kernel.
+The tasklet callback runs in atomic context, inside a software interrupt, meaning that it cannot sleep or access user-space data, so not all work can be done in a tasklet handler.
+Also, the kernel only allows one instance of any given tasklet to be running at any given time; multiple different tasklet callbacks can run in parallel.
+
+In current kernels, tasklets can be replaced by workqueues, timers, or threaded interrupts.
```

Comment:
Should "recent kernels" rather than "current kernels" according to the context.

Reply:
It is the first time that "threaded interrupt" is mentioned. You should add footnote such as
> The goal of threaded interrupts is to push more of the work to separate threads, so that the minimum needed for acknowledging an interrupt is reduced, and therefore the time spent handling the interrupt (where it can't handle any other interrupts at the same time) is reduced.
> See https://lwn.net/Articles/302043/

Reply:
In case you didn't know about threaded interrupt, you can check the slides: https://elinux.org/images/e/ef/InterruptThreads-Slides_Anderson.pdf
I was discussing on threaded interrupts and RT as well. See https://hackmd.io/@sysprog/preempt-rt